dask/bytes/local.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import print_function, division, absolute_import
+
+from glob import glob
+import logging
+import os
+import sys
+
+from .compression import files as compress_files, seekable_files
+from .utils import SeekableFile
+from ..base import tokenize
+from ..compatibility import FileNotFoundError
+from ..delayed import delayed
+from ..utils import read_block, system_encoding
+
+logger = logging.getLogger(__name__)
+
+
+def read_bytes(fn, delimiter=None, not_zero=False, blocksize=2**27,
+        sample=True, compression=None):
+    """ See dask.bytes.core.read_bytes for docstring """
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+    if '*' in fn:
+        filenames = list(map(os.path.abspath, sorted(glob(fn))))
+        sample, first = read_bytes(filenames[0], delimiter, not_zero,
+                                blocksize, sample=True, compression=compression)
+        rest = [read_bytes(f, delimiter, not_zero, blocksize, sample=False,
+                           compression=compression)[1] for f in filenames[1:]]
+        return sample, [first] + rest
+    else:
+        if not os.path.exists(fn):
+            raise FileNotFoundError(fn)
+
+        if blocksize is None:
+            offsets = [0]
+        else:
+            size = getsize(fn, compression)
+
+            offsets = list(range(0, size, blocksize))
+            if not_zero:
+                offsets[0] = 1
+
+        token = tokenize(fn, delimiter, blocksize, not_zero, compression,
+                         os.path.getmtime(fn))
+
+        logger.debug("Read %d blocks of binary bytes from %s", len(offsets), fn)
+        f = delayed(read_block_from_file)
+
+        values = [f(fn, offset, blocksize, delimiter, compression,
+                    dask_key_name='read-file-block-%s-%d' % (token, offset))
+                  for offset in offsets]
+
+        if sample:
+            if isinstance(sample, int) and not isinstance(sample, bool):
+                nbytes = sample
+            else:
+                nbytes = 10000
+            sample = read_block_from_file(fn, 0, nbytes, None, compression)
+
+        return sample, values
+
+
+def read_block_from_file(fn, offset, length, delimiter, compression):
+    with open(fn, 'rb') as f:
+        if compression:
+            f = SeekableFile(f)
+            f = compress_files[compression](f)
+        result = read_block(f, offset, length, delimiter)
+        if compression:
+            f.close()
```

Comment:
Do compression wrappers leave open (temporary) files hanging around? If you need to close here, then an error in read_block (incorrect argument types) would mean you don't get here.